### ISSUE TITLE
Reporting Dashboard: Status of each Operation's Report 

### ISSUE THREAD

**Issue by mikevespi** [bcgov/cas-reporting#219]
#### Description:

Reports will have a set of statuses that represent their current state. 
Currently, there are:
- Not started
- Draft
- Submitted

[Wireframe](https://www.figma.com/file/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?type=design&node-id=7789%3A2031&mode=design&t=mcPbilWJY8esC5XQ-1)

#### Acceptance Criteria:

Given I am an Industrial Reporter on the Reporting Dashboard Page
When I have NOT yet started an Annual Report for an Operation
Then the status for that report will be "Not Started" 

Given I am an Industrial Reporter on the Reporting Dashboard Page
When I have STARTED (and saved) an Annual Report for an Operation
Then the status for that report will be "Draft" 

Given I am an Industrial Reporter on the Reporting Dashboard Page
When I have COMPLETED and submitted an Annual Report for an Operation
Then the status for that report will be "Submitted" 

#### Development Checklist:

- [x] Create a status column cell rendering file ~  `ReportingOperationStatusCell.tsx` similar to `bciers/apps/administration/app/components/userOperators/cells/UserOperatorStatusCell.tsx` 
- [x] In file  `ReportingOperationStatusCell.tsx` add logic for cell styling and param.value text display:
```
Not started
Draft
Submitted
```
- [x] In file `bciers/apps/reporting/src/app/components/datagrid/models/operations/operationColumns.tsx`, add column "Status"=`report_status` before  "Actions" column using  `renderCell: ReportingOperationStatusCell` (similar to column "Status"=`status` in file `bciers/apps/administration/app/components/datagrid/models/userOperators/userOperatorColumns.ts`)
- [x] In file `bciers/apps/reporting/src/app/components/datagrid/models/operations/operationGroupColumns.ts` define `report_status` as searchable (similar to file `bciers/apps/administration/app/components/datagrid/models/userOperators/userOperatorGroupColumns.ts`)
```
{
      groupId: "report_status",
      headerName: "Status",
      renderHeaderGroup: SearchCell,
      children: [{ field: "report_status" }],
    },
```
- [ ] Meets the DOD

~The `report_last_edited_by/timestamp` values are computed by looking at the various elements of the report and taking the most recent one. Unless we want to explore an extra report model informed by signals?~

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-
#### Blocked By

#258

**Comment by rdromey** [comment]
@patriciarussellCAS to finish writing AC and @shon-button to write dev checklist

**Comment by patriciarussellCAS** [comment]
AC complete.

**Comment by patriciarussellCAS** [comment]
Let's prioritize this one as it will be part of the UAT and control testing.

**Comment by patriciarussellCAS** [comment]
Added the High Priority label because this was flagged as a "medium" risk by MNP.

**Comment by fviduya** [comment]
I have tested this in the DEV environment and Statuses show accordingly. [Here's](https://bcgov-my.sharepoint.com/:w:/g/personal/francesca_viduya_gov_bc_ca/EdzawEYe79NMor_Oou-DYBYBdgSrxNkQy1RbyqPXHd5_Vg?e=oRukte) the test document for reference.

cc: @vesselak / @patriciarussellCAS
